interfaces/builtin/shared_memory.go

```diff
@@ -0,0 +1,155 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+const sharedMemorySummary = `allows two snaps to use predefined shared memory objects`
+
+const sharedMemoryBaseDeclarationSlots = `
+  shared-memory:
+    allow-installation:
+      slot-snap-type:
+        - app
+    allow-connection:
+      plug-attributes:
+        shared-memory: $SLOT(shared-memory)
+    allow-auto-connection:
+      plug-publisher-id:
+        - $SLOT_PUBLISHER_ID
+      plug-attributes:
+        shared-memory: $SLOT(shared-memory)
+`
+
+// sharedMemoryInterface allows sharing sharedMemory between snaps
+type sharedMemoryInterface struct{}
+
+func (iface *sharedMemoryInterface) Name() string {
+	return "shared-memory"
+}
+
+func (iface *sharedMemoryInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              sharedMemorySummary,
+		BaseDeclarationSlots: sharedMemoryBaseDeclarationSlots,
+
+		AffectsPlugOnRefresh: true,
+	}
+}
+
+func (iface *sharedMemoryInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
+	sharedMemory, ok := slot.Attrs["shared-memory"].(string)
+	if !ok || len(sharedMemory) == 0 {
+		if slot.Attrs == nil {
+			slot.Attrs = make(map[string]interface{})
+		}
+		// content defaults to "slot" name if unspecified
+		slot.Attrs["shared-memory"] = slot.Name
+	}
+
+	pathError := func(value interface{}) error {
+		return fmt.Errorf(`shared-memory "write" attribute must be a list of strings (current value: %v)`, value)
+	}
+	// check that we have a write path
+	writeAttr, ok := slot.Attrs["write"]
+	if !ok {
+		return pathError("unset")
+	}
+
+	pathList, ok := writeAttr.([]interface{})
+	if !ok || len(pathList) == 0 {
+		return pathError(pathList)
+	}
+
+	for _, pathAttr := range pathList {
+		path, ok := pathAttr.(string)
+		if !ok {
+			return pathError(pathAttr)
+		}
+		path = strings.TrimSpace(path);
```

Comment:
Ah our good ol' friend, the terminating semi-colon :slightly_smiling_face: 
```suggestion
		path = strings.TrimSpace(path)
```